srml/scored-pool/src/lib.rs

```diff
@@ -0,0 +1,368 @@
+
+	/// The receiver of the signal for when the members have changed.
+	type MembershipChanged: ChangeMembers<Self::AccountId>;
+
+	/// Allows a configurable origin type to set a score to a candidate in the pool.
+	type ScoreOrigin: EnsureOrigin<Self::Origin>;
+
+	/// Required origin for removing a member (though can always be Root).
+	/// Configurable origin which enables removing an entity. If the entity
+	/// is part of the `Members` it is immediately replaced by the next
+	/// highest scoring candidate, if available.
+	type KickOrigin: EnsureOrigin<Self::Origin>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait<I>, I: Instance=DefaultInstance> as Membership {
+		/// The current pool of candidates, stored as an ordered Vec
+		/// (ordered ascending by score, `None` first, highest last).
+		Pool get(pool) config(): Vec<(T::AccountId, Option<T::Score>)>;
+
+		/// The current membership, stored as an ordered Vec.
+		Members get(members): Vec<T::AccountId>;
+
+		/// Size of the `Members` set.
+		MemberCount get(member_count) config(): u32;
+	}
+	add_extra_genesis {
+		config(members): Vec<T::AccountId>;
+		config(phantom): sr_std::marker::PhantomData<I>;
+		build(|
+			storage: &mut (sr_primitives::StorageOverlay, sr_primitives::ChildrenStorageOverlay),
+			config: &Self,
+		| {
+			sr_io::with_storage(storage, || {
+				let mut pool = config.pool.clone();
+
+				// reserve balance for each candidate in the pool.
+				// panicking here is ok, since this just happens one time, pre-genesis.
+				pool
+					.iter()
+					.for_each(|(who, _)| {
+						T::Currency::reserve(&who, T::CandidateDeposit::get())
+							.expect("balance too low");
+					});
+
+				<Module<T, I>>::sort_pool(&mut pool);
+				<Pool<T, I>>::put(&pool);
+				<Module<T, I>>::refresh_members(false);
+
+				let members = <Members<T, I>>::get();
+				T::MembershipInitialized::initialize_members(&members);
+			});
+		})
+	}
+}
+
+decl_event!(
+	pub enum Event<T, I=DefaultInstance> where
+		<T as system::Trait>::AccountId,
+	{
+		/// The given member was removed; see the transaction for who.
+		MemberRemoved,
+		/// An entity has issued a candidacy.
+		CandidateAdded,
+		/// An entity withdrew candidacy.
+		CandidateWithdrew,
+		/// The candidacy was forcefully removed for an entity.
+		CandidateKicked,
+		/// Phantom member, never used.
+		Dummy(sr_std::marker::PhantomData<(AccountId, I)>),
+	}
+);
+
+decl_module! {
+	pub struct Module<T: Trait<I>, I: Instance=DefaultInstance>
+		for enum Call
+		where origin: T::Origin
+	{
+		fn deposit_event<T, I>() = default;
+
+		/// Every `Period` blocks the `Members` set is refreshed from the
+		/// highest scoring members in the pool.
+		fn on_initialize(n: T::BlockNumber) {
+			if n % T::Period::get() == Zero::zero() {
+				<Module<T, I>>::refresh_members(true);
+			}
+		}
+
+		/// Add `origin` to the pool of candidates.
+		pub fn issue_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+
+			let _ = Self::find_in_pool(&who)
```

Comment:
Why not just: 

```suggestion
			let _ = Self::find_in_pool(&who)? 
```

The function, I think, has the same return type as the dispatch.

But I'd actually recommend this because it is more _srml-idiomatic_:

```suggestion
			ensure!(Self::find_in_pool(&who).is_err(), "already a member");
```